lib/evss/disability_compensation_auth_headers.rb

```diff
@@ -31,7 +31,7 @@ def eauth_json
     end
 
     def gender
-      case @user.gender || @user.mpi&.profile&.gender
```

Comment:
@bramleyjl this change will negatively impact users.  It was intentional to fall back to MPI's gender when gender was nil from the identity provider.  Can you please revert it?

Reply:
The `@user.gender` function that this leverages currently falls back to the MPI gender. Although there is a check for `mhv_icn` to use the MPI Profile gender, so there is a slight difference. Would that be applicable in this case?

Reply:
no, the mhv_icn is only present when MHV is the identity provider. if the gender is nil in ID.me or DS logon this change will make the request fail.